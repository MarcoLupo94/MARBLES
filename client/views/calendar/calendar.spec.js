import React from 'react';
import renderer from 'react-test-renderer';
import CalendarScroll from './calendar';
import { screen, render } from '@testing-library/react';

describe('Calendar component', () => {
  test('Calendar renders correctly', () => {
    const tree = renderer.create(<CalendarScroll />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Calendar have 3 children', () => {
    const tree = renderer.create(<CalendarScroll />).toJSON();
    expect(tree.children.length).toBe(2);
  });
  test('Should render', () => {
    render(<CalendarScroll />);
    screen.getByTestId('list');
  });
});
