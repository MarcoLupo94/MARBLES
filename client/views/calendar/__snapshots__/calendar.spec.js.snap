// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar component Calendar renders correctly 1`] = `
<RCTSafeAreaView
  data-testid="list"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#7197AC",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          "2022-05-17T19:57:24.944Z",
          "2022-06-17T19:57:24.944Z",
          "2022-07-17T19:57:24.944Z",
          "2022-08-17T19:57:24.944Z",
          "2022-09-17T19:57:24.944Z",
          "2022-10-17T19:57:24.944Z",
          "2022-11-17T20:57:24.944Z",
          "2022-12-17T20:57:24.944Z",
          "2023-01-17T20:57:24.944Z",
          "2023-02-17T20:57:24.944Z",
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      getItemLayout={[Function]}
      horizontal={false}
      initialNumToRender={3}
      initialScrollIndex={6}
      keyExtractor={[Function]}
      nestedScrollEnabled={true}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEnabled={true}
      scrollEventThrottle={50}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {
            "backgroundColor": "white",
          },
          undefined,
        ]
      }
      testID="undefined.list"
      viewabilityConfigCallbackPairs={
        Array [
          Object {
            "onViewableItemsChanged": [Function],
            "viewabilityConfig": Object {
              "viewAreaCoveragePercentThreshold": 20,
            },
          },
        ]
      }
    >
      <View>
        <View
          style={null}
        >
          <Text
            style={
              Array [
                Array [
                  Object {
                    "minHeight": 360,
                    "width": 750,
                  },
                  Object {
                    "paddingLeft": 15,
                    "paddingRight": 15,
                  },
                  undefined,
                ],
                Object {
                  "color": "#2d4150",
                  "fontSize": 20,
                  "fontWeight": "200",
                },
              ]
            }
          >
            2022-05-17
          </Text>
        </View>
        <View
          style={null}
        >
          <Text
            style={
              Array [
                Array [
                  Object {
                    "minHeight": 360,
                    "width": 750,
                  },
                  Object {
                    "paddingLeft": 15,
                    "paddingRight": 15,
                  },
                  undefined,
                ],
                Object {
                  "color": "#2d4150",
                  "fontSize": 20,
                  "fontWeight": "200",
                },
              ]
            }
          >
            2022-06-17
          </Text>
        </View>
        <View
          style={null}
        >
          <Text
            style={
              Array [
                Array [
                  Object {
                    "minHeight": 360,
                    "width": 750,
                  },
                  Object {
                    "paddingLeft": 15,
                    "paddingRight": 15,
                  },
                  undefined,
                ],
                Object {
                  "color": "#2d4150",
                  "fontSize": 20,
                  "fontWeight": "200",
                },
              ]
            }
          >
            2022-07-17
          </Text>
        </View>
        <View
          style={null}
        >
          <Text
            style={
              Array [
                Array [
                  Object {
                    "minHeight": 360,
                    "width": 750,
                  },
                  Object {
                    "paddingLeft": 15,
                    "paddingRight": 15,
                  },
                  undefined,
                ],
                Object {
                  "color": "#2d4150",
                  "fontSize": 20,
                  "fontWeight": "200",
                },
              ]
            }
          >
            2022-08-17
          </Text>
        </View>
        <View
          style={null}
        >
          <Text
            style={
              Array [
                Array [
                  Object {
                    "minHeight": 360,
                    "width": 750,
                  },
                  Object {
                    "paddingLeft": 15,
                    "paddingRight": 15,
                  },
                  undefined,
                ],
                Object {
                  "color": "#2d4150",
                  "fontSize": 20,
                  "fontWeight": "200",
                },
              ]
            }
          >
            2022-09-17
          </Text>
        </View>
        <View
          style={null}
        >
          <Text
            style={
              Array [
                Array [
                  Object {
                    "minHeight": 360,
                    "width": 750,
                  },
                  Object {
                    "paddingLeft": 15,
                    "paddingRight": 15,
                  },
                  undefined,
                ],
                Object {
                  "color": "#2d4150",
                  "fontSize": 20,
                  "fontWeight": "200",
                },
              ]
            }
          >
            2022-10-17
          </Text>
        </View>
        <View
          style={null}
        >
          <Text
            style={
              Array [
                Array [
                  Object {
                    "minHeight": 360,
                    "width": 750,
                  },
                  Object {
                    "paddingLeft": 15,
                    "paddingRight": 15,
                  },
                  undefined,
                ],
                Object {
                  "color": "#2d4150",
                  "fontSize": 20,
                  "fontWeight": "200",
                },
              ]
            }
          >
            2022-11-17
          </Text>
        </View>
        <View
          style={null}
        >
          <Text
            style={
              Array [
                Array [
                  Object {
                    "minHeight": 360,
                    "width": 750,
                  },
                  Object {
                    "paddingLeft": 15,
                    "paddingRight": 15,
                  },
                  undefined,
                ],
                Object {
                  "color": "#2d4150",
                  "fontSize": 20,
                  "fontWeight": "200",
                },
              ]
            }
          >
            2022-12-17
          </Text>
        </View>
        <View
          style={null}
        >
          <Text
            style={
              Array [
                Array [
                  Object {
                    "minHeight": 360,
                    "width": 750,
                  },
                  Object {
                    "paddingLeft": 15,
                    "paddingRight": 15,
                  },
                  undefined,
                ],
                Object {
                  "color": "#2d4150",
                  "fontSize": 20,
                  "fontWeight": "200",
                },
              ]
            }
          >
            2023-01-17
          </Text>
        </View>
        <View
          style={
            Object {
              "height": 360,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#7197AC",
        "bottom": 0,
        "display": "absolute",
        "flexDirection": "row",
        "height": "10%",
        "justifyContent": "space-around",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/Calendar.png",
          }
        }
        style={
          Object {
            "height": 35,
            "width": 35,
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/Update.png",
          }
        }
        style={
          Object {
            "height": 35,
            "width": 35,
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/LogOut.png",
          }
        }
        style={
          Object {
            "height": 35,
            "width": 30,
          }
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
